src/tests/multihost/ad/test_login_attr.py

```diff
@@ -0,0 +1,336 @@
+            'ldap_schema': 'ad',
+            'ldap_user_principal': 'userPrincipalName',
+        }
+        client.sssd_conf(dom_section, sssd_params)
+        client.sssd_conf('sssd', {'services': 'nss, pam, ssh'})
+        client.clear_sssd_cache()
+
+        # Create AD user
+        (aduser, _) = create_aduser_group
+
+        # Search for the user
+        usr_cmd = multihost.client[0].run_command(
+            f'getent passwd {aduser}',
+            raiseonerr=False
+        )
+
+        # Run su command
+        su_result = client.su_success(aduser)
+        # Run ssh command
+        ssh_result = client.auth_from_client(aduser, 'Secret123') == 3
+
+        client.restore_sssd_conf()
+        client.clear_sssd_cache()
+
+        # Evaluate test results
+        assert usr_cmd.returncode == 0, f"User {aduser} was not found."
+        assert su_result, "The su command failed!"
+        assert ssh_result, "The ssh login failed!"
+
+    @staticmethod
+    def test_0002_login_by_userprincipalname(multihost, create_aduser_group):
+        """test_0002_login_by_userprincipalname
+
+        :title: IDM-SSSD-TC: ad_provider: ad_login_attr: Verify user login
+         with userPrincipalName
+        :id: 589f4fb3-ef62-42c1-827a-77d73129b879
+        :setup:
+          1. Configure sssd with ldap_user_principal = userPrincipalName
+        :steps:
+          1. Run su for the user with fully qualified name.
+          2. Run ssh for the user with fully qualified name.
+        :expectedresults:
+          1. Su command succeeds.
+          3. Ssh command succeeds.
+        :customerscenario: False
+        """
+        # Configure sssd
+        multihost.client[0].service_sssd('stop')
+        ad_realm = multihost.ad[0].domainname.upper()
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        client.backup_sssd_conf()
+        dom_section = f'domain/{client.get_domain_section_name()}'
+        sssd_params = {
+            'ldap_id_mapping': 'True',
+            'debug_level': '9',
+            'id_provider': 'ad',
+            'use_fully_qualified_names': 'False',
+            'ad_domain': multihost.ad[0].domainname.lower(),
+            'ad_server': multihost.ad[0].hostname,
+            'ldap_schema': 'ad',
+            'ldap_user_principal': 'userPrincipalName',
+        }
+        client.sssd_conf(dom_section, sssd_params)
+        client.sssd_conf('sssd', {'services': 'nss, pam, ssh'})
+        client.clear_sssd_cache()
+
+        # Create AD user
+        (aduser, _) = create_aduser_group
+
+        # Run su command
+        su_result = client.su_success(f'{aduser}@{ad_realm}')
+        # Run ssh command
+        ssh_result = client.auth_from_client(
+            f'{aduser}@{ad_realm}', 'Secret123') == 3
+        client.restore_sssd_conf()
+        client.clear_sssd_cache()
+
+        # Evaluate test results
+        assert su_result, "The su command failed!"
+        assert ssh_result, "The ssh login failed!"
```

Comment:
Asserts do not match with expectedresults from docstring

Reply:
The message behind the assert is error message to show in the case of failure so they actually match.